drivers/led_strip/tlc5971.c

```diff
@@ -0,0 +1,363 @@
+
+/**
+ * @brief GS reference clock edge select bit for OUTXn on-off timing control in FC data
+ */
+#define TLC5971_CTRL_BIT_OUTTMG (BIT(1))
+
+/**
+ * @brief Constant-current output enable bit in FC data (0 = output control enabled, 1 = blank).
+ */
+#define TLC5971_CTRL_BIT_BLANK (BIT(5))
+
+/**
+ * @brief Auto display repeat mode enable bit in FC data (0 = disabled, 1 = enabled).
+ */
+#define TLC5971_CTRL_BIT_DSPRPT (BIT(6))
+
+/**
+ * @brief Display timing reset mode enable bit in FC data (0 = disabled, 1 = enabled).
+ */
+#define TLC5971_CTRL_BIT_TMGRST (BIT(7))
+
+/**
+ * @brief Bit mask for control bits in data byte 27
+ */
+#define TLC5971_BYTE27_CTRL_MASK (TLC5971_CTRL_BIT_EXTGCK | TLC5971_CTRL_BIT_OUTTMG)
+
+/**
+ * @brief Bit mask for control bits in data byte 26
+ */
+#define TLC5971_BYTE26_CTRL_MASK                                                                   \
+	(TLC5971_CTRL_BIT_BLANK | TLC5971_CTRL_BIT_DSPRPT | TLC5971_CTRL_BIT_TMGRST)
+
+/**
+ * @brief create data byte 27 from control data
+ *
+ * @param control_data control bits
+ * @return uint8_t the serialized data byte 27
+ */
+static inline uint8_t tlc5971_data_byte27(uint8_t control_data)
+{
+	return (TLC5971_WRITE_COMMAND << 2) | (control_data & TLC5971_BYTE27_CTRL_MASK);
+}
+
+/**
+ * @brief create data byte 26 from control data and color 1 GBC
+ *
+ * @param control_data control bits
+ * @param gbc_color_1 global brightness control for color 1 LEDs
+ * @return uint8_t the serialized data byte 26
+ */
+static inline uint8_t tlc5971_data_byte26(uint8_t control_data, uint8_t gbc_color_1)
+{
+	return (control_data & TLC5971_BYTE26_CTRL_MASK) | (gbc_color_1 >> 2);
+}
+
+/**
+ * @brief create data byte 25 from color 1 and 2 GBC
+ *
+ * @param gbc_color_1 global brightness control for color 1 LEDs
+ * @param gbc_color_2 global brightness control for color 2 LEDs
+ * @return uint8_t the serialized data byte 25
+ */
+static inline uint8_t tlc5971_data_byte25(uint8_t gbc_color_1, uint8_t gbc_color_2)
+{
+	return (gbc_color_1 << 6) | (gbc_color_2 >> 1);
+}
+
+/**
+ * @brief create data byte 24 from color 2 and 3 GBC
+ *
+ * @param gbc_color_2 global brightness control for color 2 LEDs
+ * @param gbc_color_3 global brightness control for color 3 LEDs
+ * @return uint8_t the serialized data byte 24
+ */
+static inline uint8_t tlc5971_data_byte24(uint8_t gbc_color_2, uint8_t gbc_color_3)
+{
+	return (gbc_color_2 << 7) | gbc_color_3;
+}
+
+/**
+ * @brief map user colors to tlc5971 color order
+ *
+ * @param color_id color id from color mapping
+ * @param pixel_data rgb data to be mapped
+ * @return the mapped color value
+ */
+static uint8_t tlc5971_map_color(int color_id, const struct led_rgb *pixel_data)
+{
+	uint8_t temp = 0;
+
+	switch (color_id) {
+	case LED_COLOR_ID_RED:
+		temp = pixel_data->r;
+		break;
+	case LED_COLOR_ID_GREEN:
+		temp = pixel_data->g;
+		break;
+	case LED_COLOR_ID_BLUE:
+		temp = pixel_data->b;
+		break;
+	default:
+		temp = 0;
```

Comment:
Default is white? If so please use `LED_COLOR_ID_WHITE`

Reply:
White is not supported by the led strip API so I don't think it makes sense to use that here. I also removed it from the init function

Reply:
Okay. In that case, you have already initialized `temp` with `0`, so no need to do it again here.